moped-editor/src/views/projects/newProjectView/NewProjectView.js

```diff
@@ -242,13 +236,18 @@ const NewProjectView = () => {
    */
   const timer = useRef();
 
-  useEffect(() => {
-    const currentTimer = timer.current;
+  /**
+   * saveActionState contains the current save state
+   * saveActionDispatch allows us to update the state via action (signal) dispatch.
+   */
+  const { saveActionState, saveActionDispatch } = useSaveActionReducer();
 
-    return () => {
-      clearTimeout(currentTimer);
-    };
-  }, []);
+  const handleSubmitDispatch = () => {
+    // Check this is not already done
+    if (saveActionState?.currentStep === 0) {
+      saveActionDispatch({ type: "initiateFeatureSave" });
+    }
+  };
```

Comment:
Whenever we hit "Save" starts the process by dispatching a "initiateFeatureSave" signal